api/app/controllers/spree/api/customer_returns_controller.rb

```diff
@@ -62,6 +63,30 @@ def load_order
       def customer_return_params
         params.require(:customer_return).permit(permitted_customer_return_attributes)
       end
+
+      def build_return_items_from_params
+        customer_return_attributes = customer_return_params
+        return_items_params = customer_return_attributes.
+          delete(:return_items_attributes)
+
+        if return_items_params.is_a? ActionController::Parameters
+          return_items_params = return_items_params.values
+        end
+
+        @customer_return = CustomerReturn.new(customer_return_attributes)
+
+        @customer_return.return_items = return_items_params.map do |item_params|
+          return_item = if item_params[:id]
+                          Spree::ReturnItem.find(item_params[:id])
+                        else
+                          Spree::ReturnItem.new
+                        end
+
+          return_item.assign_attributes(item_params)
+
+          return_item
+        end.compact
```

Comment:
I'm not sure why we need `compact` after the `map` cycle, I suppose we're always returning a `Spree::ReturnItem` with each iteration, and when a return item cannot be found from `item_params[:id]` an  exception is raised... but maybe I'm overlooking some scenario?

Also, I think the naming of the method reflects only part of what this method does here. What about a more general `build_customer_return`, since it's actually building the instance stored in`@customer_return`?

Reply:
> I'm not sure why we need compact after the map cycle
I agree, upon closer inspection I don't think the `.compact` is necessary here.

> I think the naming of the method reflects only part of what this method does here.
That's a great suggestion, I think `build_customer_return` is a better name here, I will update that!

Thanks for the feedback, I will update this change to reflect the above suggestions!

Reply:
@spaghetticode I've addressed your comments here and this should be ready for another review 🙏🏼